forge-game/src/main/java/forge/game/player/Player.java

```diff
@@ -625,6 +625,13 @@ public final boolean payLife(final int lifePayment, final SpellAbility cause, fi
             return false;
         }
 
+        // Ashiok Exile instead of paying life
+        if (hasKeyword("Exile library instead of pay life") && lifePayment <= getZone(ZoneType.Library).size()) {
```

Comment:
while technically you can't get different results when controlling multiple Ashiok it should probably still be done as Replacement to avoid problems later?

Reply:
Ashiok, Wicked Manipulator doesn't have "exiled with"
But probably yeah 

I will think about that

Of cause that makes passing the params even more complicated